go.mod

```diff
@@ -1,6 +1,14 @@
 module sigs.k8s.io/kind
 
-go 1.16
+// NOTE: This is the go language version, NOT the compiler version.
+//
+// This controls the *minimum* required go version and therefore available Go
+// language features.
+//
+// See ./.go-version for the go compiler version used when building binaries
+//
+// https://go.dev/doc/modules/gomod-ref#go
+go 1.17
```

Comment:
well , this is changing in 1.21

> For example, a go.mod that says go 1.21.0 with no toolchain line is interpreted as if it had a toolchain go1.21.0 line.

https://go.dev/doc/toolchain

Reply:
We should rework everything around .go-version to switch to this toolchain line.

Reply:
Thanks, I forgot about this.

Reply:
so, I added a commit to deal with this, but the remaining downside (Also true in k/k currently), is that IDEs won't see all our bash hacks.

what we probably need to do is pivot to setting toolchain in the go mods and then also for the image builds of external binaries, but that's going to force us to decentralize setting the go version unfortunately.

Reply:
the current PR is enough to ensure our builds / tools work at least